scripts/zap-scanner.sh

```diff
@@ -37,60 +34,74 @@ elif [ "$ENVIRONMENT" = "local" ]; then
 else
     echo "Invalid environment $ENVIRONMENT"
     exit 1
+fi
+
+cd "$TARGET_DIR" || exit 2
 
+# Ensure the APP_URL is reachable from the zaproxy container
+if ! docker-compose run --rm zaproxy curl -Is "$APP_URL" > /dev/null 2>&1; then
```

Comment:
From the `zaproxy` docker container, confirm the application can be reached before initializing the full zap scan.